tests/test_configuration.py

```diff
@@ -799,6 +824,24 @@ def setUpClass(cls):
 	def tearDownClass(cls):
 		sys.path.pop(0)
 
+class TestBar(TestCase):
+	def test_bar(self):
+		import powerline as powerline_module
+		with swap_attributes(config, powerline_module):
+			with get_powerline_raw(config, powerline_module, ext='wm', renderer_module='bar') as powerline:
+				self.assertEqual("%{{l}}{}%{{r}}{}".format(powerline.render(side='left'), powerline.render(side='right')), "SOME_STRING_HERE")
+
+	def test_bar_escape(self):
+		from powerline.shell import ShellPowerline
+		import powerline as powerline_module
+		config['themes/wm/default']['segments']['left'] = {
```

Comment:
https://docs.python.org/2.7/reference/expressions.html#set-displays. Set literal is `"{" expression_list "}"` part, a tuple or a list should work here. Set literals may actually also work… if only dictionaries were hashable and only by a chance because order is not guaranteed. They are not and you get `TypeError` when trying to construct set literal with the output of `highlighted_string`.